src/query/Index.hpp

```diff
@@ -14,11 +14,13 @@ namespace precice {
 namespace query {
 
 /// Struct to hold index and distance information of the closest primitive
+/// @todo Replace default magic number initialization with a dedicated variable
 template <class Tag>
 struct MatchType {
   double distance;
   int    index;
```

Comment:
This struct is a good candidate for [aggregate initialization](https://en.cppreference.com/w/cpp/language/aggregate_initialization).
In this case, we would need to delete all constructors and then specify default member initializers.

```cpp
struct MatchType {
  double distance{-1};
  int    index{-1};
  constexpr bool operator<(MatchType const &other) const;
};
```

Reply:
I agree on that, but do not know enough about aggregate initialization. Are you sure we need to remove all constructors (and why can we)?
What I would do now would look the following way:
```cpp
template <class Tag>
struct MatchType {
  double distance{-1};
  int    index{-1};
  MatchType() = default;
  MatchType(double d, int i)
      : distance(d), index(i){};

  constexpr bool operator<(MatchType const &other) const
  {
    return distance < other.distance;
  };
};
````

Reply:
The compiler writes the constructors for you. Have a look at [this example](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:2,endLineNumber:16,positionColumn:2,positionLineNumber:16,selectionStartColumn:2,selectionStartLineNumber:16,startColumn:2,startLineNumber:16),source:'%23include%3Ciostream%3E%0A%0Astruct+S+%7B%0A++++int+a%7B-1%7D%3B%0A++++double+b%7B-1%7D%3B%0A%7D%3B%0A%0Aint+main()+%7B%0A++++S+s1%3B%0A++++std::cout+%3C%3C+s1.a+%2B+s1.b+%3C%3C+!'%5Cn!'%3B%0A%0A++++S+s2%7B1,+0.3%7D%3B+//+aggregate+init%0A++++std::cout+%3C%3C+s2.a+%2B+s2.b+%3C%3C+!'%5Cn!'%3B%0A%0A++++return+0%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:35.77023498694517,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:clang_trunk,filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'0',intel:'0',libraryCode:'0',trim:'1'),fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!((name:eigen,ver:'339')),options:'-Wall+-Wextra+--std%3Dc%2B%2B14',selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'x86-64+clang+(trunk)+(Editor+%231,+Compiler+%231)+C%2B%2B',t:'0')),k:50,l:'4',m:35.56581986143187,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compiler:1,editor:1,fontScale:14,fontUsePx:'0',wrap:'1'),l:'5',n:'0',o:'%231+with+x86-64+clang+(trunk)',t:'0')),header:(),l:'4',m:64.43418013856814,n:'0',o:'',s:0,t:'0')),k:64.22976501305483,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4).

Reply:
I see, thanks. I will add it, although it will hinder people from using `MatchType(1,2,)` with round braces `()` meaning this only works for curly braces, but I'm fine with that.

Reply:
I cannot yet apply this change in a nice way due to the following warning (I skipped the following one):
```bash
~/precice/src/query/Index.cpp:45:115: warning: narrowing conversion of ‘matchID’ from ‘size_t’ {aka ‘long unsigned int’} to ‘int’ [-Wnarrowing]
   45 |                                        match = VertexMatch{bg::distance(sourceCoord, _mesh->vertices()[matchID]), matchID};
``` 
This is again related to #248. I added now the aggregate initialization without actually using it, meaning I still have the constructors in the class. 

Since I promised to finish this today, I will merge here. If you don't like it or think it is worth changing the aggregate initialization stuff in `Index.cpp`, feel free to add it later on.